composer.json

```diff
@@ -26,6 +25,7 @@
         "bref/bref": "^0.5.14"
     },
     "require-dev": {
+        "aws/aws-sdk-php": "^3.127",
```

Comment:
Why move it to require-dev?

Reply:
I want to make a follow up PR to introduce Async AWS. See https://github.com/brefphp/symfony-messenger/pull/16#discussion_r395148988

With this factories and the TransportProvider we have detached ourselves from `aws/aws-sdk-php`. 

Yes, it is a BC break. But Im okey with this since nothing worked "out of the box" before, you still had to register a `Aws\Sqs\SqsClient` service.

Reply:
OK I see. I'm on the fence because this brings more complexity, and this mixes with a PR that isn't directly related.

What if this package always used async-aws? Would that work? From what I can picture in my head, this package can have its own SQS/SNS client and that wouldn't leak out to impact the user 🤔 

I'd rather avoid a situation where this package is compatible with both SDKs, because that means more maintenance effort and more effort for users as well.